packages/netlify-cms-backend-github/src/API.js

```diff
@@ -459,14 +475,55 @@ export default class API {
     return metadata;
   };
 
+  async migrateToVersion1(oldContentKey, metaData) {
+    const newContentKey = this.generateContentKey(metaData.collection, oldContentKey);
+    const newBranchName = this.generateBranchName(newContentKey);
+
+    // create new branch and pull request in new format
+    const newBranch = await this.createBranch(newBranchName, metaData.pr.head);
+    const pr = await this.createPR(metaData.commitMessage, newBranchName);
+
+    // store new metadata
+    await this.storeMetadata(newContentKey, {
+      ...metaData,
+      pr: {
+        number: pr.number,
+        head: pr.head.sha,
+      },
+      branch: newBranchName,
+      version: '1',
+    });
+
+    // remove old data
+    await this.closePR(metaData.pr);
+    await this.deleteBranch(metaData.branch);
+    await this.deleteMetadata(oldContentKey);
+
+    return newBranch;
+  }
+
+  async migrateBranch(branch) {
+    const contentKey = this.contentKeyFromRef(branch.ref);
+    const metadata = await this.retrieveMetadata(contentKey);
+    if (!metadata.version) {
+      // migrate branch from cms/slug to cms/collection/slug
+      branch = await this.migrateToVersion1(contentKey, metadata);
+    }
+
+    return branch;
+  }
+
   async listUnpublishedBranches() {
     console.log(
       '%c Checking for Unpublished entries',
       'line-height: 30px;text-align: center;font-weight: bold',
     );
-    const onlyBranchesWithOpenPRs = filterPromisesWith(({ ref }) =>
-      this.branchHasPR({ branchName: this.branchNameFromRef(ref), state: 'open' }),
-    );
+    const onlyBranchesWithOpenPRs = flowAsync([
```

Comment:
Used a new `flowAsync` utility function, which is cleaner.

Reply:
Please see here:
https://github.com/netlify/netlify-cms/commit/b37faa0a95b87d0a1b45ac026bdc09720802342f#diff-b4ba42c688a23ed9cec5139daec2c157R518

Instead of sending a `pull` API request per branch and since that API uses substring match, we can send a single request with `cms` as the matching pattern and filter the branches according to the result.